### DISCUSSION TITLE
Github actions permissions issue after Subscription Change - Help requested

### DISCUSSION THREAD

**Discussion by mattgu74** [community/community#76409] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello all,

This weekend, I enabled github enterprise trial but it kill all our github actions workflows, we cancel the trial to come back to the team subscription level and see if it's better, but that's not.

Actually all the workflows stopped because of permissions issues, like : 
``` 
[Invalid workflow file: .github/workflows/ci.yml#L24](https://github.com/<redacted>/<redacted>/actions/runs/6931844739/workflow)
The workflow is not valid. .github/workflows/ci.yml (Line: 24, Col: 3): Error calling workflow '<redacted>/actions/.github/workflows/build-docker-image.yml@v2'. The nested job 'build-docker-image' is requesting 'id-token: write', but is only allowed 'id-token: none'.
```

I guess something that need to be optin on old organizations, has been automatically opt-in while changing subscriptions level.
Is there a way to opt-out for this settings ? Because we have so many workflows to fix because of that, that this is really depressing :/ 

Thanks a lot for your answers,
Matthieu

**Comment by mattgu74** [comment]
I finally found the settings by myself. In organization settings, there is a Workflow permissions option.

**Reply by hendricius** [reply, marked as answer]
That fixed it thanks. Just to everyone having the same issue. These are the settings in the current UI:

Open this menu in the repo settings (I don't have an org):

<img width="377" alt="image" src="https://github.com/community/community/assets/816859/1a078f79-fa99-4c03-9796-06bb42c057b2">

Then tick this option:
<img width="984" alt="image" src="https://github.com/community/community/assets/816859/da6a2b41-35f3-4a61-a532-bf7922aaf299">

**Reply by xuhdev** [reply]
I think this is an unnecessarily broad elevation of permissions. The permission control can be narrowed to the specific workflow by elevating the permissions from the caller's side. See my comment below https://github.com/orgs/community/discussions/76409#discussioncomment-11260212

**Reply by samus-aran** [reply]
I will mark this as the answer but I'm sure folks find @xuhdev suggestion helpful to narrow down permissions to specific workflows.

For more information, refer to the [GitHub Documentation on Workflow Permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token).

**Comment by TobiasUhmannVoss** [comment]
I have the same problem and could change the default permissions in the organization settings. But I don't want to. I want to have the restricted default permissions and make use of the YAML's job "permissions". And the "Workflow permissions" section in both the organization and repo settings states that "You can specify more granular permissions in the workflow using YAML.". Apparently, I can not. Furthermore, I checked the docs that "specifying more granular permissions" is not limited to further restricting permissions. According to the docs it should be possible to elevate permissions:

"You can modify the permissions for the GITHUB_TOKEN in individual workflow files. If the default permissions for the GITHUB_TOKEN are restrictive, you may have to elevate the permissions to allow some actions and commands to run successfully." [1]

Is there something different with organization default permissions? Are they an upper permission limit?

[1] https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

**Comment by TobiasUhmannVoss** [comment]
Okay, so I'm not sure if this is a bug, but I think it has something to do with me using nested workflows. This is my setup:

main workflow
  main job 1 (calls sub workflow 1)
  main job 2 (calls sub workflow 2)

sub workflow 2:
  sub job 1
  sub job 2 (needs permissions)

When requesting the permissions for "sub job 2" directly, I get the aforementioned error. When requesting them for the "main job 2", I do not. This is still not exactly what I want but much better than relaxing the organizational settings.

**Comment by xuhdev** [comment]
The safe way to do this is to elevate the permissions when calling the workflow:

```yml
jobs:
  call-workflow:
    permissions:
      id-token: write
  uses: org/workflows/.github/workflows/workflow.yml@<sha256>
```

See https://github.com/8hobbies/react-autonumeric/blob/bb7ac1f5f29138a80d5d00f2e51c32acb673ce29/.github/workflows/pages.yml#L25-L27 for a live example.
